components/eamxx/src/physics/mam/readfiles/fractional_land_use.hpp

```diff
@@ -0,0 +1,95 @@
+#ifndef FRACTIONAL_LANDUSE_HPP
+#define FRACTIONAL_LANDUSE_HPP
+
+// For AtmosphereInput
+#include "share/io/scorpio_input.hpp"
+
+namespace scream {
+namespace frac_landuse {
+
+template <typename ScalarType, typename DeviceType>
+struct fracLandUseFunctions {
+  using Device = DeviceType;
+
+  using KT      = KokkosTypes<Device>;
+  using view_2d = typename KT::template view_2d<Real>;
+
+  // -------------------------------------------------------------------------------------------
+  // -------------------------------------------------------------------------------------------
+
+  struct FracLandUseData {
+    FracLandUseData() = default;
+    FracLandUseData(const int ncol_, const int nclass_) {
+      init(ncol_, nclass_, true);
+    }
+    void init(const int ncol_, const int nclass_, const bool allocate) {
+      ncols  = ncol_;
+      nclass = nclass_;
+      if(allocate) frac_land_use = view_2d("FRAC_LAND_USE", ncols, nclass);
+    }
+    // Basic spatial dimensions of the data
+    int ncols;
+    int nclass;
+
+    // Fractional land use (unitless) dimensions = (nclass, ncols)
+    view_2d frac_land_use;
+  };  // FracLandUseData
+
+  // -------------------------------------------------------------------------------------------
+  // -------------------------------------------------------------------------------------------
+
+  struct FracLandUseInput {
```

Comment:
Since you have no vert remap, you don't need this auxiliary data structure. I would remove it, and just use the tgt field out of the remapper.

Reply:
I have removed both data structures.